### ISSUE TITLE
Probably-unintended metric RegEx in SageMaker notebook

### ISSUE THREAD

**Issue by athewsey** [huggingface/notebooks#61]
Hi all & thanks for the examples!

I see that in [SageMaker notebook 6](https://github.com/huggingface/notebooks/blob/master/sagemaker/06_sagemaker_metrics/sagemaker-notebook.ipynb) some metrics are set up as follows:

```python
metric_definitions=[
    {'Name': 'loss', 'Regex': "'loss': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'learning_rate', 'Regex': "'learning_rate': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_loss', 'Regex': "'eval_loss': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_accuracy', 'Regex': "'eval_accuracy': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_f1', 'Regex': "'eval_f1': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_precision', 'Regex': "'eval_precision': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_recall', 'Regex': "'eval_recall': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_runtime', 'Regex': "'eval_runtime': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_samples_per_second', 'Regex': "'eval_samples_per_second': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'epoch', 'Regex': "'epoch': ([0-9]+(.|e\-)[0-9]+),?"}]
```

I had an issue today with the way `learning_rate` was picking up (missing the exponent) and realised that the RegEx's used here probably aren't doing what's intended.

- They only match **either** a decimal point `.123` **or** an exponent `e-123`, not both (e.g. in a learning rate like `1.03e-6`
- ...And in fact you *have* to have exactly one of those (so no integer metrics e.g. `42`)
- They only match negative exponents (which granted is the usual use case, but maybe not for every metric?)
- They only match positive numbers (which granted is the usual use case, but maybe not for every metric?)
- We have unescaped backslashes and haven't indicated it's a [raw string](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) via e.g. `r"'loss': ...\..."`

If careful validation is the aim, maybe it could be something more like the following?

```python
r"'my_metric': (-?[0-9]+(\.[0-9]+)?(e[-+]?[0-9]+)?),?"
```

...Or perhaps we could be a little more concise and trusting on the numbers, with something like the below?

```python
r"'my_metric': ([-+0-9e.]+)[,}]"
```

As long as the expression is able to articulate what comes immediately after the number (always comma or close brace as far as I can tell?), we could even be super lazy and e.g. `'my_metric': (.*?)[,}]`.

**Comment by philschmid** [comment]
@athewsey Thank you for identifying this feel free to open a PR.
